text/3389-manifest-lint.md

```diff
@@ -0,0 +1,278 @@
+
+Override the default level of lints by assigning them to a new level in a
+table, for example:
+```toml
+[lints]
+unsafe = "forbid"
+```
+
+Supported levels include:
+- `forbid`
+- `deny`
+- `warn`
+- `allow`
+
+**Note:** TOML does not support `:` in unquoted keys, requiring tool-specific
+lints to be quoted, like
+```toml
+[lints]
+"clippy::enum_glob_use" = "warn"
+```
+
+## The `lints` table
+
+*as a new [`[workspace]` entry](https://doc.rust-lang.org/cargo/reference/workspaces.html#the-workspace-section)*
+
+The `workspace.lints` table is where you define lint configuration to be inherited by members of a workspace.
+
+Specifying a workspace lint configuration is similar to package lints.
+
+Example:
+
+```toml
+# [PROJECT_DIR]/Cargo.toml
+[workspace]
+members = ["crates/*"]
+
+[workspace.lints]
+unsafe = "forbid"
+```
+
+```toml
+# [PROJECT_DIR]/crates/bar/Cargo.toml
+[package]
+name = "bar"
+version = "0.1.0"
+
+[lints]
+workspace = true
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+When parsing a manifest, cargo will resolve workspace inheritance for
+`lints.workspace = true` as it does with basic fields, when `workspace` is
+present, no other fields are allowed to be present.  This precludes having the
+package override the workspace on a lint-by-lint basis.
+
+When running rustc, cargo will transform the lints from `lint = level` to
+`--level lint` and pass them on the command line before `RUSTFLAGS`, allowing
+user configuration to override package configuration.  These flags will be
+fingerprinted so changing them will cause a rebuild.
+
+**Note:** This reserves the lint name `workspace` to allow workspace inheritance.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+A concern brought up in
+[rust-lang/rust-clippy#1313](https://github.com/rust-lang/rust-clippy/issues/1313)
+was that this will pass lints unconditionally to the underlying tool, leading
+to "undefined lint" warnings when used on earlier versions, requiring that
+warning to also be suppressed, reducing its value.  However, in the "Future
+possibility's", we mention direct support for tying lints to rust versions.
+
+This does not allow sharing lints across workspaces.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+This could be left to `clippy.toml` but that leaves `rustc` without a solution.
+
+`[lints]` could be `[package.lints]`, tying it to the package unlike `[patch]`
+and other fields that are more workspace related.  Instead, we used
+`[dependencies]` as our model.
+
+`[lints]` could be `[lint]` but we decided to follow the precedence of `[dependencies]`.
+
+Instead of using `::` as a separator between tool and lint (e.g.
+`clippy::enum_glob_use`), we could use TOML dotted keys for this (e.g.
+`clippy.enum_glob_use`).  This has the advantage of allowing unquoted keys at
+the cost of not being able to copy/paste the lint name from the tool's output
+into the file.
```

Comment:
I think we should do this; the subtable model seems like a perfect fit for tool lints, and the syntax feels much easier to work with without having to quote the names (which already makes it not a clean copy and paste).

Reply:
RFC has been updated.  I was originally waiting to see where other threads went but figured I'd take a stab at it now and I can change it later.